com.unity.sg2/Tests/HeadlessPreview/AllNodesTest.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using NUnit.Framework;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor.ShaderGraph.GraphDelta;
+using System.Linq;
+using UnityEditor.ShaderGraph.HeadlessPreview.UnitTests;
+using UnityEngine.TestTools;
+using static UnityEditor.VersionControl.Message;
+
+namespace UnityEditor.ShaderGraph.HeadlessPreview.NodeTests
+{
+
+    // TODO: Consider async compile/rendering, need to wait until thread is done and results are returned
+    // https://forum.unity.com/threads/support-for-async-await-in-tests.787853/#post-6838493
+
+    [TestFixture]
+    class DefaultPreviewsTestFixture
+    {
+        static ShaderGraphRegistry InitSGR()
+        {
+            ShaderGraphRegistry SGR = new();
+            SGR.InitializeDefaults();
+            return SGR;
+        }
+
+        static string[] InitNodeNames()
+        {
+            return SGR.DefaultTopologies.GetNodes().Select(e => e.ID.LocalPath).ToArray();
+        }
+
+        // whether or not a node will require another node to coerce the value to a vector for testing preview compilation.
+        static bool HasVectorOutput(NodeHandler node)
+        {
+            var nodeKey = node.GetRegistryKey();
+            var typeField = node.GetPorts().Where(e => !e.IsInput)?.First()?.GetTypeField();
+            bool hasPreviewableOutputPort = typeField?.GetRegistryKey().Name == GraphType.kRegistryKey.Name;
+            return hasPreviewableOutputPort && GraphTypeHelpers.GetHeight(typeField) == GraphType.Height.One;
+        }
+
+        static ShaderGraphRegistry SGR = InitSGR();
+        // Need to resolve the node names statically so that the Test Runner is happy and shows each node.
+        static string[] nodeNames = InitNodeNames();
+
+        [TestCaseSource("nodeNames")]
+        public void DoesPreviewCompile(string nodeName)
+        {
+            Registry Registry = SGR.Registry;
+            GraphHandler Graph = new(Registry);
+
+            HeadlessPreviewManager Preview = new();
+            Preview.SetActiveRegistry(Registry);
+            Preview.SetActiveGraph(Graph);
+            Preview.Initialize("ThisDontMatter", new UnityEngine.Vector2(125, 125));
+
+            var nodeKey = SGR.DefaultTopologies.GetNode(nodeName).GetRegistryKey();
+            var node = Graph.AddNode(nodeKey, nodeName);
+
+            string previewName = nodeName;
+            if (!HasVectorOutput(node))
+            {
+                var outPort = node.GetPorts().Where(e => !e.IsInput)?.First();
+                var portName = outPort.LocalID;
+                var typeName = outPort?.GetTypeField()?.GetRegistryKey().Name;
+                previewName = "Helper";
+
+                if (typeName == GraphType.kRegistryKey.Name)
+                {
+                    var keyMatDet = new RegistryKey { Name = "MatrixDeterminant", Version = 1 };
```

Comment:
Are these `RegistryKeys` hard coded values for nodes from the current implementation? If so, is there a way to get the node from the registry that does what you want without having a hard coded name/version?

Reply:
IStandardNode would need to expose the RegistryKey, then yes-- it could be pulled statically.

Reply:
I did move these definitions out to a `static readonly` space to make it easier to swap them out when it's possible to do so.

Reply:
Ah, yes. Okay. I think that `StandardNode` should do this. I'll make a task to do that and update these tests.

Reply:
Here is the task in jira for this https://jira.unity3d.com/browse/GSG-1190. It links back to this PR conversation thread.

Reply:
I'm fine with closing this conversation when you are.